infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/WaitSnapshotApplyState.java

```diff
@@ -159,6 +172,16 @@ public LogReplicationState processEvent(LogReplicationEvent event) throws Illega
         }
     }
 
+    private void deleteForcedSyncRequestFromEventTable() {
+        CorfuStoreEntry<ReplicationEventInfoKey, ReplicationEvent, Message> eventEnqueuedForSession =
+                fsm.getAckReader().getMetadataManager().getEventEnqueuedForSession(fsm.getSession());
+
+        if (eventEnqueuedForSession.getPayload() != null && eventEnqueuedForSession.getPayload().getEventId().equals(transitionSyncId.toString())) {
```

Comment:
If the original forced sync encounters a SYNC_CANCEL and gets re-triggered with a new syncId this condition would not be hit right? We would need to keep the id stored in the event table updated

Reply:
Thanks for catching this Shreay.
We have the implementation to enqueue the sync_cancel event with the snapshotID of the cancelled snapshot sync. We were missing the part where we use the syncId from the event to trigger a new snapshot sync ID.

I have implemented that fix in the recent push. So now, if a snapshot sync gets cancelled in either SnapshotSync state or WaitSnapshotApply state, the new snapshot sync triggered will have the same syncID as that of the cancelled sync. Only for the snapshot syncs triggered from log_entry sync would have a different sync ID as that we would the start of a new sync cycle.